### PR TITLE
[5.0] Fixes the cli installer

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue #42114 .

### Summary of Changes

- refactor the code so cli installation doesn't throw 

### Testing Instructions

Install joomla using `cd installation` and `php joomla.php install`.
Make sure that you set the public folder option `--public-folder` to a valid directory (or do not provide any options and follow the questions but provide a valid path for the public folder)

### Actual result BEFORE applying this Pull Request

DB error

### Expected result AFTER applying this Pull Request

No errors

### Link to documentations
Please select:
- [x] No documentation changes for docs.joomla.org needed
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: libraries/src/Helper/PublicFolderGeneratorHelper.php [hunk 1]
```diff
@@ -161,19 +161,18 @@ public function createPublicFolder(string $destinationPath): void
...
         $this->createFile($fullDestinationPath . 'api/index.php', str_replace(['{{APPLICATIONPATH}}', '{{DEFINESPATH}}'], ['\'' . DIRECTORY_SEPARATOR . 'api\'', 'dirname(__DIR__)'], $this->indexTemplate));
 
-        if (is_dir(JPATH_ROOT . '/images')) {
-            $this->createSymlink($root . 'images', $destinationPath . 'images', JPATH_ROOT . '/');
-        }
-
-        // Create symlinks for all the local filesystem directories
-        if (PluginHelper::isEnabled('filesystem', 'local')) {
+        // Get all the local filesystem directories
+        if (defined('_JCLI_INSTALLATION')) {
```
Comment (richard67) [inline libraries/src/Helper/PublicFolderGeneratorHelper.php]:
```suggestion
        if (\defined('_JCLI_INSTALLATION')) {
```
Fix PHPCS failing in drone.

Comment (dgrammatiko) [reply libraries/src/Helper/PublicFolderGeneratorHelper.php]: Did it already but thanks

**Comment by richard67** [comment]
@dgrammatiko Shouldn't the testing instructions tell to use the public folder option `--public-folder`? As far as I understood the issue, the CLI installation broke only with using that option. Or was that wrong and it's broken also when being called without that opzion?

**Comment by dgrammatiko** [comment]
> Shouldn't the testing instructions tell to use the public folder option --public-folder?

Good call, fixed

**Comment by richard67** [comment]
> > Shouldn't the testing instructions tell to use the public folder option --public-folder?
> 
> Good call, fixed

@dgrammatiko Where? I see no change in the description.

**Comment by dgrammatiko** [comment]
@richard67 fwiw `php joomla.php install` without any options kicks an interactive set of questions, so the last question is about the public folder path. Leaving it empty skips the ceremony for the public folder...

**Comment by hans2103** [comment]
@dgrammatiko 
This is what I see on my local machine at the end of `cd installation;php joomla.php install` before applying the PR.
I do not have a DB error. 

```
Relative or absolute path to the public folder []:
 > 

OK
Validating DB connection...OK
Creating and populating the database...OK
Writing configuration.php and additional setup ...OK

                                                                                                                        
 [OK] Joomla has been installed                                                                                         
                                                                                                                        

hans2103@192 installation % 
```

**Comment by rabidgrowth** [comment]
Thank for the PR! Confirm is work now! ✅ 

Hope this includes in 5.0 at next Tuesday. I will use this + Ansible. Make build server easy for me.

**Comment by HLeithner** [comment]
this pr is scheduled for 5.0.1 thanks @rabidgrowth for the test

**Comment by rabidgrowth** [comment]
@ceford You put all joomla files inside the folder /Users/ceford/Sites/j5rc2test, yes? Can not use same folder /Users/ceford/Sites/j5rc2test for public. Must use different folder. Example: /Users/ceford/Sites/rabidgrowth. After install ends the site is servered from /Users/ceford/Sites/rabidgrowth. 

See my issue https://github.com/joomla/joomla-cms/issues/42114 . I used folder outside from /var/www for joomla files, then folder inside /var/www for public. 

Hope this is help for you!

**Comment by dgrammatiko** [comment]
@hans2103 @ceford the CLI installation is broken only if the user provided a path for the public folder (and maybe the path needs to be outside of the root path) otherwise nothing is broken

**Comment by ceford** [comment]
I see now, the cli installer is used for creating a site within a site or multiple sites on the same server. That did not occur to me. Thanks for the clarification.

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on ec5c8d2cef7a65d28770a54ace0dfb6e04873379<br /><br />To test I unpacked the download in ~/tmp/joomla5 and followed the instructions. Without the patch I got the Database error and with the patch I got a working site in ~/Sites/j5testcli (I keep all my test sites in ~/Sites). I had not twigged this is a way to keep most of the php code outside the public folder.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42116">issues.joomla.org/tracker/joomla-cms/42116</a>.</sub>

**Comment by richard67** [comment]
> Thank for the PR! Confirm is work now! ✅

@rabidgrowth Could you mark your test result right? It's not enough just to add a comment with a green check mark. It needs to submit your test result in the issue tracker. For doing this, go to https://issues.joomla.org/tracker/joomla-cms/42116 , then use the blue "Test this" at the top left corner, then select your test result and finally submit. Thanks in advance.

**Comment by rabidgrowth** [comment]
I have tested this item :white_check_mark: successfully on ec5c8d2cef7a65d28770a54ace0dfb6e04873379<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42116">issues.joomla.org/tracker/joomla-cms/42116</a>.</sub>

**Comment by rabidgrowth** [comment]
Sorry, didn't know. I now marked test inside tracker.

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42116">issues.joomla.org/tracker/joomla-cms/42116</a>.</sub>

**Comment by richard67** [comment]
@ceford @rabidgrowth Thanks for testing.

**Comment by HLeithner** [comment]
thanks, there is another problem I will create a PR for

**Comment by dgrammatiko** [comment]
@HLeithner https://github.com/joomla/joomla-cms/issues/42120 or another one?

**Comment by HLeithner** [comment]
I talk about #42168 no fix for #42120 yet

**Comment by dgrammatiko** [comment]
Nice, but I think the error msg needs to be translatable

**Pull request merged**